### PR TITLE
pip-zipapp: New package for bootstrapping python pipeline

### PR DESCRIPTION
**Pull request by pnasrat**
See #20318 needed to bootstrap python pipeline

#### For new package PRs only

- [ ] This PR is part of python toolchain improvements
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ x REQUIRED - The version of the package is still receiving security updates
- [ n/a ] This PR links to the upstream project's support policy (e.g. `endoflife.date`) (see main pip package)

### CODE REVIEW

**Review by smoser — DISMISSED** [review]

File: pip-zipapp.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  description: "Pip as a python zipapp"
+  copyright:
+    - license: MIT
+
+environment:
+  contents:
+    packages:
+      - busybox
+
+pipeline:
```
Comment (vaikas) [inline pip-zipapp.yaml:14]: Could we add some comments for where the various sources below are fetched from, and why. Not being a python expert, it's a bit tricky to reason about it. Things like if there's an update to pypa/pip, how do these get updated.

File: pip-zipapp.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      cp "pip-${{package.version}}.pyz" "$destd/$sdir/$fname"
+      chmod 0644 "$destd/$sdir/$fname"
+
+      mkdir -p "$destd/$sdir/wheels"
+      cp *.whl "$destd/$sdir/wheels/"
+
+update:
+  enabled: true
+  github:
+    identifier: pypa/pip
```
Comment (vaikas) [inline pip-zipapp.yaml:65]: Looks like there are beta releases, so wonder if they should be excluded too?
https://github.com/pypa/pip/tags

Comment (pnasrat) [reply pip-zipapp.yaml:76]: @vaikas done

File: pip-zipapp.yaml (hunk 1 above)
Comment (pnasrat) [reply pip-zipapp.yaml:23]: Done, I'll add an issue once merged to see if we can make this clearer

**Comment by pnasrat** [comment]
Also filed https://github.com/wolfi-dev/wolfictl/issues/960 to make wolfictl support full semver pre-releases

**Review by smoser — APPROVED** [review]
looks good.

**Pull request merged**